python/src/aiconfig/default_parsers/claude.py

```diff
@@ -0,0 +1,297 @@
+        params: Optional[dict[Any, Any]] = {},
+    ) -> dict[Any, Any]:
+        """
+        Defines how to parse a prompt in the .aiconfig for a particular model
+        and constructs the completion params for that model.
+
+        Args:
+            Update this documentation... serialized_data (str): Serialized data from the .aiconfig.
+
+        Returns:
+            dict: Model-specific completion parameters.
+        """
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_deserialize_start",
+                __name__,
+                {"prompt": prompt, "params": params},
+            )
+        )
+        # Build Completion params
+        model_settings = self.get_model_settings(prompt, aiconfig)
+
+        completion_data = refine_chat_completion_params(
+            model_settings, aiconfig, prompt
+        )
+
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+
+        # Claude is trained using RLHF, need to add the human prompt to the beginning of the prompt
+        # See https://docs.anthropic.com/claude/docs/introduction-to-prompt-design#human--assistant-formatting
+        completion_data[
+            "prompt"
+        ] = f"{HUMAN_PROMPT} {resolved_prompt}{AI_PROMPT}"
+
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_deserialize_complete",
+                __name__,
+                {"output": completion_data},
+            )
+        )
+
+        return completion_data
+
+    async def run_inference(
+        self,
+        prompt: Prompt,
+        aiconfig: "AIConfigRuntime",
+        options: Optional[InferenceOptions] = None,
+        parameters: Dict[Any, Any] = {},
+    ) -> List[Output]:
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_run_start",
+                __name__,
+                {
+                    "prompt": prompt,
+                    "options": options,
+                    "parameters": parameters,
+                },
+            )
+        )
+
+        if self.client is None:
+            # AWS credentials could either be in the environment or in ~/.aws/credentials
+            # Let Anthropic's API handle AWS credentials validation which happens on Api call, not on client construct
+            self.client = AnthropicBedrock()
+
+        completion_data = await self.deserialize(prompt, aiconfig, parameters)
+
+        # if stream enabled in runtime options and config, then stream. Otherwise don't stream.
+        stream = True  # Default value
+        if options is not None and options.stream is not None:
+            stream = options.stream
+        elif "stream" in completion_data:
+            stream = completion_data["stream"]
+
+        completion_data["stream"] = stream
+
+        response = self.client.completions.create(**completion_data)  # type: ignore (pyright doesn't understand response object)
```

Comment:
API call will throw If aws credentials are not set. I removed mine locally and tested this manually.